src/x/ident/tag_iterator.go

```diff
@@ -139,6 +140,10 @@ func (i *tagSliceIter) Err() error {
 }
 
 func (i *tagSliceIter) Close() {
+	if i.closed {
+		return
+	}
+	i.closed = true
```

Comment:
Will need to set closed = false for both Reset and ResetFields I believe so will close properly after the two types of resets are used/called?

Reply:
ugh. good catch. will update the test.